src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -134,14 +132,48 @@ Transaction::pointer Transaction::load(uint256 const& id, Application& app)
 
         *db << sql, soci::into (ledgerSeq), soci::into (status),
                 soci::into (sociRawTxnBlob, rti);
-        if (!db->got_data () || rti != soci::i_ok)
-            return {};
 
-        convert(sociRawTxnBlob, rawTxn);
+        auto const got_data = db->got_data ();
+
+        if ((!got_data || rti != soci::i_ok) && !useRange)
+            return nullptr;
+
+        if (!got_data)
```

Comment:
May as well check `rti` here too.

Reply:
I don't bother checking `rti` here since when the SQL statement fails to retrieve data `rti != soci::i_ok` is always true.